src/Controller/OcrController.php

```diff
@@ -216,9 +229,9 @@ private function getText(): string
             ]
```

Comment:
Now that the result includes translated messages, I think we need the interface language to be part of the cache key, otherwise the warning messages won't be translated for subsequent requests. e.g. the following should return different results, but are both in English:

```bash
{pr/49} ~/code/wikimedia/wikimedia-ocr$ curl -s 'http://localhost:8000/api?image=http%3A%2F%2Flocalhost%2Fwikimedia%2Fwiki1%2Fimages_wiki1%2Fthumb%2Fa%2Faf%2FTestdoc.pdf%2Fpage2-1024px-Testdoc.pdf.jpg&engine=tesseract&psm=3&lang=asdfasd&uselang=en' | jq .warnings
[
  "The following languages are invalid or not supported by the engine and were ignored: asdfasd"
]
{pr/49} ~/code/wikimedia/wikimedia-ocr$ curl -s 'http://localhost:8000/api?image=http%3A%2F%2Flocalhost%2Fwikimedia%2Fwiki1%2Fimages_wiki1%2Fthumb%2Fa%2Faf%2FTestdoc.pdf%2Fpage2-1024px-Testdoc.pdf.jpg&engine=tesseract&psm=3&lang=asdfasd&uselang=qqx' | jq .warnings
[
  "The following languages are invalid or not supported by the engine and were ignored: asdfasd"
]
```

It might be better to add the warnings after retrieving the cached data, to avoid extra cache items (although I don't really think it's likely to be that much of a problem: we probably don't serve the same page in different interface languages all that often, especially with invalid language codes).

Reply:
Good point, done! I went with adding the interface language to the cache key. An alternative might be to store msg key + params in EngineResult->warnings, and then localize them in the OcrController, which is I think what you meant by "add the warnings after retrieving the cached data". I think that would require a few changes though, e.g. for proper handling of array parameters, which should be run through listToText etc., and I'm also unsure if we're going to need that level of precision. Do you think the current solution (i.e. varying on the interface language) is enough for now?

Reply:
Yep, keeps it simple! Looks good to me.